### PR TITLE
Source Azure Table: update use of ConnectorStateManager

### PR DESCRIPTION
**Pull request by clnoll**
Depends on CDK updates in https://github.com/airbytehq/airbyte/pull/34540.

The `ConnectorStateManager` has been updated to take a mapping of stream name: `AirbyteStream` instead of name: `Stream` instance. This updates the connectors that use the `ConnectorStateManager` to conform to this change.

TODO: update to real version of CDK once https://github.com/airbytehq/airbyte/pull/34540 has been released.

### CODE REVIEW

**Review by girarda — APPROVED** [review]
approved pending the release of https://github.com/airbytehq/airbyte/pull/34540 and updating the dependencies to point to the latest cdk version (or >=)

File: docs/integrations/sources/azure-table.md [hunk 1]
```diff
@@ -66,7 +66,8 @@ We recommend creating a restricted key specifically for Airbyte access. This wil
 ## Changelog
 
 | Version | Date       | Pull Request                                             | Subject                                           |
-| :------ | :--------- | :------------------------------------------------------- | :------------------------------------------------ |
+|:--------|:-----------| :------------------------------------------------------- |:--------------------------------------------------|
+| 0.1.4   | 2024-01-31 | [15591](https://github.com/airbytehq/airbyte/pull/15591) | Update use of ConnectorStateManager               |
```
Comment (natikgadzhi) [inline docs/integrations/sources/azure-table.md:70]:
```suggestion
| 0.1.4   | 2024-01-31 | [34716](https://github.com/airbytehq/airbyte/pull/34716) | Update use of ConnectorStateManager               |
```

**Review by natikgadzhi — APPROVED** [review]
Nice! Does this mean that Azure Table (and other file-based connectors using latest CDK) gets incremental concurrent syncs out of the box?

File: docs/integrations/sources/azure-table.md (hunk 1 above)
Comment (clnoll) [reply docs/integrations/sources/azure-table.md:70]: Thank you!

**Review by bazarnov — APPROVED** [review]

**Comment by clnoll** [comment]
Instead of this change, we've decided to simply update `ConnectionStateManager` to take either `AirbyteStream` or `Stream` for the time being, since both offer the methods used by `ConnectionStateManager`.

In https://github.com/airbytehq/airbyte/issues/35023 we'll refactor `ConnectionStateManager` and usages to be more streamlined.

@natikgadzhi to answer your question: the file-based sources can use the Concurrent CDK now, but this PR isn't required for that. I'm in the process of writing up some notes on how to migrate connectors to it.

**Pull request closed without merging**